src/adapter/src/optimize/peek.rs

```diff
@@ -236,28 +240,38 @@ impl Optimize<MirRelationExpr> for Optimizer {
 impl LocalMirPlan<Unresolved> {
     /// Produces the [`LocalMirPlan`] with [`ResolvedLocal`] contextual
     /// information required for the next stage.
-    pub fn resolve(
+    pub fn resolve<'a>(
         self,
-        session: &Session,
+        session: &'a Session,
         stats: Box<dyn StatisticsOracle>,
-    ) -> LocalMirPlan<ResolvedLocal> {
+        peek_ctx: &'a PeekContext,
+    ) -> LocalMirPlan<ResolvedLocal<'a>> {
```

Comment:
I think you _should_ be able to again rely on lifetime elision here now that there's only one lifetime in play.

```suggestion
   pub fn resolve(
         self,
         session: &Session,
         stats: Box<dyn StatisticsOracle>,
         peek_ctx: &PeekContext,
     ) -> LocalMirPlan<ResolvedLocal> {
```

Reply:
I had that originally before doing the two lifetime thing. It's giving the error 
```
missing lifetime specifier
this function's return type contains a borrowed value, but the signature does not say 
whether it is borrowed from `session` or `peek_ctx`
```

Reply:
Ah ok, never mind—this is as simple as it gets then!